### PR TITLE
Collection expressions: check conversion to ICollection<T> for spread optimization

### PR DESCRIPTION
**Pull request by cston**
Use `Conversions` method to check for `ICollection<T>` for spread optimization.

Fixes https://github.com/dotnet/roslyn/issues/74894

See https://github.com/dotnet/roslyn/pull/74630
See https://github.com/dotnet/roslyn/pull/74769#issuecomment-2308489340

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs:1080]:
```suggestion
                            if (enumeratorInfo.GetEnumeratorInfo.Method.ReturnType.IsValueType)
```

File: src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs:1083]: nit: since the below conversion.Kind check for the AddRangeMethod parameter type is very similar in purpose, I would prefer if they were written in the same way, revising one or both as desired.

**Review by RikkiGibson — APPROVED** [review]

**Comment by cston** [comment]
@dotnet/roslyn-compiler for a second review, thanks.

File: src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs
Comment (chsienki) [inline src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs:1078]: Nit: 
```suggestion
                            // If collection has a struct enumerator but doesn't have a conversion to ICollection<T>
```

**Review by chsienki — APPROVED** [review]

File: src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs
Comment (cston) [reply src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs:1078]: Thanks. I'll hold off changing the comment for now because changes here will conflict with another PR in progress, and since I believe the existing comment captures the intent. We can update later if needed.

**Pull request merged**
